src/screens/HomeStack/TransactionStack/transaction-address/transaction-address.tsx

```diff
@@ -26,109 +29,39 @@ import {NetworkProviderTypes} from '@app/services/backend';
     return true;
   }, [navigation]);
-  const {from, to, nft, token} = useTypedRoute<
-    TransactionStackParamList,
-    TransactionStackRoutes.TransactionAddress
-  >().params;
+
+  const {toAddress, fromAddress, toChainId} = TransactionStore;
+
   const {bottom: safeAreaBottomInset} = useSafeAreaInsets();
 
-  const [address, setAddress] = useState(to ?? '');
   const [isError, setIsError] = useState(false);
 
-  const fromWallet = useMemo(() => Wallet.getById(from)!, [from]);
+  const fromWallet = useMemo(() => Wallet.getById(fromAddress)!, [fromAddress]);
 
   const [loading, setLoading] = React.useState(false);
 
-  const filteredWallets = useMemo(() => {
-    const wallets = Wallet.getAllVisible();
-
-    if (!wallets?.length) {
-      return [];
-    }
-    const isTron = Provider.selectedProvider.isTron;
-
-    if (!address) {
-      return wallets.filter(w => {
-        if (isTron && !w.isSupportTron) {
-          return false;
-        }
-        return !AddressUtils.equals(w.address, from);
-      });
-    }
-
-    const lowerCaseAddress = address.toLowerCase();
-
-    return wallets.filter(w => {
-      if (isTron && !w.isSupportTron) {
-        return false;
-      }
-      return (
-        (w.address.toLowerCase().includes(lowerCaseAddress) ||
-          w.tronAddress?.toLowerCase?.()?.includes?.(lowerCaseAddress) ||
-          w.cosmosAddress.toLowerCase().includes(lowerCaseAddress) ||
-          w.name.toLowerCase().includes(lowerCaseAddress)) &&
-        !AddressUtils.equals(w.address, from)
-      );
-    });
-  }, [address, from]);
-
-  const filteredContacts = useMemo(() => {
-    const contacts = Contact.getAll();
-
-    if (!contacts?.length) {
-      return [];
-    }
-
-    if (!address) {
-      return contacts.filter(c => !AddressUtils.equals(c.account, from));
-    }
-
-    const lowerCaseAddress = address.toLowerCase();
-
-    return contacts.filter(c => {
-      const hexAddress = AddressUtils.toEth(c.account);
-      const haqqAddress = AddressUtils.toHaqq(hexAddress);
-
-      return (
-        (hexAddress.includes(lowerCaseAddress) ||
-          haqqAddress.includes(lowerCaseAddress) ||
-          c.name?.toLowerCase().includes(lowerCaseAddress)) &&
-        !AddressUtils.equals(hexAddress, from)
-      );
-    });
-  }, [address, from]);
-
   const onDone = useCallback(
     async (result: string) => {
-      // const isTron = Provider.selectedProvider.isTron;
-      // if (contacts?.length === 0 && filteredWallets?.length === 1) {
-      //   return onDone(
-      //     isTron ? filteredWallets[0].tronAddress : filteredWallets[0].address,
-      //   );
-      // }
-
-      // if (contacts?.length === 1 && filteredWallets?.length === 0) {
-      //   return onDone(contacts[0].account);
-      // }
-
-      // if (AddressUtils.isValidAddress(address.trim())) {
-      //   return onDone(address.trim());
-      // }
-
       try {
         const networkType = Provider.selectedProvider.isTron
```

Comment:
refactor Provider.selectedProvider to TransactionStore

Reply:
Done